### ISSUE TITLE
train_adapter.py — validation code is dead/non-functional (LM head never actually loaded)

### ISSUE THREAD

**Issue by bong-water-water-bong** [1bit-MONSTER/1bit-MONSTER#606]
## Summary

The validation section of `tools/train_adapter.py` attempts to compute top-1 token match rates by loading an LM head from a `.trg` file, but the LM head loading code is incomplete and never actually executes — the validation reduces to reporting a stale cosine similarity value instead of computing actual token accuracy.

## Location

`tools/train_adapter.py`, lines 138-172 (the "Validation: top-1 token match" section):

```python
print("\n── Validation: top-1 token match ──")
model.eval()
with torch.no_grad():
    # Compute logits for a few samples
    import subprocess
    # We need the LM head to compute tokens from hidden states
    # Load it from the .trg file
    lm_path = "/tmp/model.trg"
    with open(lm_path, 'rb') as f:
        hdr = f.read(512)  # skip header
        # LM head is at a specific offset
        # For now, just report similarity
        pass  # <--- NEVER ACTUALLY LOADS OR USES THE LM HEAD
    
    matches = 0
    n_test = min(100, len(X_val))
    for i in range(n_test):
        pred = model(X_val[i:i+1])
        pred_h = pred * Y_std + Y_mean
        sim_i = nn.CosineSimilarity(dim=1)(pred_h, Y_val[i:i+1] * Y_std + Y_mean).item()
    
    # BUG: uses `sim` from the *last epoch's* validation loop, NOT computed here!
    print(f"  Mean cosine sim on {n_test} samples: {sim:.4f}")
```

## The Bugs

1. **Dead code**: The `.trg` file header is read and immediately discarded. The LM head weights are never loaded, so logit-based token comparison is never performed
2. **Stale variable**: `sim` on the final `print()` call refers to the variable from the *training loop's last validation epoch*, not from this dedicated validation section. The loop above computes `sim_i` but never assigns to a new aggregate variable
3. **Hardcoded path**: `/tmp/model.trg` is hardcoded instead of being configurable
4. **Placeholder comment**: "For now, just report similarity" with a `pass` — this is a TODO that was never addressed

## Impact

- The validation output printed is **not actually measuring what it claims to measure**
- Users believe they're seeing top-1 token match accuracy when they're actually seeing a stale epoch-level cosine similarity
- Makes the training script unreliable for evaluating adapter quality

## Fix

Either:
1. Remove the broken validation block and only report cosine similarity (honest)
2. Or properly implement LM head loading from the `.trg` binary format and compute actual logit-based token matching

**Comment by bong-water-water-bong** [comment]
Fixed by #618 (merged).
